### PR TITLE
[Ray clusters][docs] Fix TOCs/index pages/other small bugs

### PR DESCRIPTION
**Pull request by cadedaniel**
## Why are these changes needed?
Fixing small details that were missed:
* Deduplicated the TOC for the `Ray Clusters>User Guides>Launching Clusters` and `Ray Clusters>User Guides>Community Supported Cluster Managers`. 
  * This might have nuked the presence of the subpages in the left nav bar, will confirm once we promote the new structure and fix if broken.
* [The Ray Clusters page](https://docs.ray.io/en/master/cluster/cluster_under_construction/getting-started.html) links to the Run Ray Jobs section that @stephanie-wang is working on.
* The [Ray Clusters for VMs index page](https://docs.ray.io/en/master/cluster/cluster_under_construction/ray-clusters-on-vms/index.html) now has a small overview and cards.
  * Adjusted the formatting so the cards look the same in the vm overview page and the k8s overview page. Both look like the Ray Clusters overview page.
* Added a TOC for the VM user guides
* Added TOC for the VM examples, linking to @jjyao's work https://docs.ray.io/en/master/cluster/cluster_under_construction/ray-clusters-on-vms/examples/index.html
* Switch the order of GCP/Azure because GCP is officially supported, in https://docs.ray.io/en/master/cluster/cluster_under_construction/ray-clusters-on-vms/user-guides/community-supported-cluster-manager/index.html
* Added note to delete https://docs.ray.io/en/master/cluster/cluster_under_construction.html

## Related issue number

N/A

### CODE REVIEW

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 1]
```diff
@@ -29,6 +29,13 @@ Ray clusters are officially supported on the following technology stacks:
 Advanced users may want to :ref:`deploy Ray clusters on-premise <on-prem>`
 or onto infrastructure platforms not listed here by :ref:`providing a custom node provider <ref-cluster-setup-under-construction>`.
 
+How can I run workloads on Ray clusters?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Workloads can be submitted to your cluster via :ref:`Ray Job Submission<ref-deployment-guide-under-construction>`.
+Ray clusters provide a dashboard to :ref:`monitor and observe<cluster-monitoring-and-observability>` the status of your application.
+When enabled, autoscaling is a powerful feature of Ray. See :ref:`Autoscaling<cluster-autoscaler-discussion>` for information on
+whether to enable autoscaling and how autoscaling works.
+
```
Comment (cadedaniel) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:38]: @stephanie-wang can I get you to look at the changes in this file? They're related to your work.

Comment (stephanie-wang) [reply doc/source/cluster/cluster_under_construction/getting-started.rst:38]: Yup, I'll modify in-place!

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 2]
```diff
@@ -29,6 +29,13 @@ Ray clusters are officially supported on the following technology stacks:
 Advanced users may want to :ref:`deploy Ray clusters on-premise <on-prem>`
 or onto infrastructure platforms not listed here by :ref:`providing a custom node provider <ref-cluster-setup-under-construction>`.
 
+How can I run workloads on Ray clusters?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Workloads can be submitted to your cluster via :ref:`Ray Job Submission<ref-deployment-guide-under-construction>`.
+Ray clusters provide a dashboard to :ref:`monitor and observe<cluster-monitoring-and-observability>` the status of your application.
+When enabled, autoscaling is a powerful feature of Ray. See :ref:`Autoscaling<cluster-autoscaler-discussion>` for information on
+whether to enable autoscaling and how autoscaling works.
```
Comment (stephanie-wang) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:37]:
```suggestion
Workloads can be submitted to your cluster via :ref:`Ray Job Submission<ref-deployment-guide-under-construction>`. Ray Jobs allow you to package your application and its dependencies for execution on a cluster.

Ray Clusters also provide tools for :ref:`monitoring and observing<cluster-monitoring-and-observability>` the status of your application.

Finally, when enabled, autoscaling is a powerful feature of Ray Clusters that automatically provisions resources for your Ray Job. See :ref:`Autoscaling<cluster-autoscaler-discussion>` for information on
whether to enable autoscaling and how autoscaling works.
```

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/index.md [hunk 3]
```diff
@@ -1,4 +1,13 @@
...
+:::{note}
+To learn the basics of Ray clusters on VMs, we recommend taking a look
+at the {ref}`introductory guide<ref-cluster-quick-start-vms-under-construction>` first.
 :::
+
+This section presents example Ray workloads to try out on your Ray cluster.
+
+More examples will be added in the future. Running the distributed XGBoost example below is a
+great way to start experimenting with production Ray workloads.
+- {ref}`clusters-vm-ml-example`
```
Comment (stephanie-wang) [inline doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/index.md:13]:
```suggestion
This section presents example Ray workloads to try out on your Ray cluster.
Running the distributed XGBoost example below is a
great way to start experimenting with production Ray workloads.
- {ref}`clusters-vm-ml-example`
```

**Review by stephanie-wang — APPROVED** [review]
Btw, are we going with "Ray cluster" instead of "Ray Cluster"?

**Comment by cadedaniel** [comment]
> Btw, are we going with "Ray cluster" instead of "Ray Cluster"?

I've been going with `Ray cluster`, wouldn't be surprised if missed something though.

**Comment by stephanie-wang** [comment]
> > Btw, are we going with "Ray cluster" instead of "Ray Cluster"?
> 
> I've been going with `Ray cluster`, wouldn't be surprised if missed something though.

Ah okay, sounds good! I've been using Ray Cluster, heh. But we can just do a find and replace at the end.

**Review by DmitriGekhtman — APPROVED** [review]

**Comment by cadedaniel** [comment]
@stephanie-wang I think I lost your changes when I force pushed after rebase. I'll add them again.

File: doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/index.md (hunk 3 above)
Comment (cadedaniel) [reply doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/examples/index.md:13]: re-added

File: doc/source/cluster/cluster_under_construction/getting-started.rst (hunk 2 above)
Comment (cadedaniel) [reply doc/source/cluster/cluster_under_construction/getting-started.rst:37]: re-added

**Comment by cadedaniel** [comment]
@ericl this should be ready for merge! leaving to you per requested feature freeze

**Pull request closed without merging**